### PR TITLE
Host-side insets for iOS UIViews

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

**Comment by swankjesse** [comment]
https://github.com/cashapp/redwood/issues/919

File: redwood-widget/src/iosMain/kotlin/app/cash/redwood/widget/RedwoodUIView.kt [hunk 1]
```diff
@@ -54,11 +55,14 @@ public open class RedwoodUIView : RedwoodView<UIView> {
   override val children: Widget.Children<UIView>
     get() = _children
 
+  private val density = Density.Default
```
Comment (JakeWharton) [inline redwood-widget/src/iosMain/kotlin/app/cash/redwood/widget/RedwoodUIView.kt]:
```suggestion
  private val density: Density get() = Density.Default
```
No need to burn a backing field on a value that never changes

**Review by JakeWharton — APPROVED** [review]

**Comment by swankjesse** [comment]
The newly-added test fails like this in CI:

```
Child process terminated with signal 5: Trace/BPT trap
```

I haven’t been able to extract more information than that. I can’t catch an exception or anything. And it passes locally. 

I assume something I’m doing with the UIWindow is failing 'cause either the code to do it isn’t there (a linking problem), or because the environment it’s running in isn’t there (headless?).

**Comment by squarejesse** [comment]
https://publicobject.com/2024/11/20/testing-safe-area-insets-on-ios/

**Pull request merged**